### PR TITLE
adds editor for post title

### PR DESCRIPTION
**Pull request by cindywu**
known problems:

- title supports next line
- title supports marks and block types
- breaks url slug when title is blank
- breaks when the url slug is already taken (not unique)

### CODE REVIEW

**Comment by cindywu** [comment]
cindy (@cindywu) : Twitter 2020-02-09 10-10-31

File: .gitignore [hunk 1]
```diff
@@ -36,4 +36,5 @@ yarn-debug.log*
 .yarn-integrity
 
 .docker-sync/*
-.docker-sync
\ No newline at end of file
+.docker-sync
+app/models/concerns/slug_history.rb
```
Comment (dennyluan) [inline .gitignore]: hm did you mean to add this in here? did you accidentally run `git rm`

**Comment by dennyluan** [comment]
I'm going to jump on this branch to finish the slug validations before merging this in.

File: .gitignore (hunk 1 above)
Comment (cindywu) [reply .gitignore]: oops! sorry. i will remove this change from this pr.

**Comment by dennyluan** [comment]
Alright I've refactored some of it and fixed the bugs with the slugging process.

For now slugs will look something like:

`abcdef-this-is-my-title`

and then keep the little 6char leader around as the title gets changed.

**Comment by cindywu** [comment]
i approve of this pr 👍🏽

**Pull request merged**